src/core/CPU.cpp

```diff
@@ -809,67 +809,164 @@ Blaze::Cycles Blaze::CPU::executePER() {
 };
 
 Blaze::Cycles Blaze::CPU::executePHA() {
-	// TODO
+	// TODO TO CHECK
+	if (memoryAndAccumulatorAre8Bit()) {
+		store8(SP, A.load());
+	}
+	else {
+		store16(SP, A.forceLoadFull());
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHB() {
-	// TODO
+	// TODO TO CHECK
+	if (e != 0) {
+		store8(SP, (1u << 8) | DBR);
+	} else {
+		store16(SP, DBR);
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHD() {
-	// TODO
+	// TODO TO CHECK
+	if (e != 0) {
+		store8(SP, DR);
+	} else {
+		store16(SP, DR);
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHK() {
-	// TODO
+	// TODO  TO CHECK
+	if (e != 0) {
+		store8(SP, (1u << 8) | PBR);
+	} else {
+		store16(SP, PBR);
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHP() {
-	// TODO
+	// TODO TO CHECK
+	if (e != 0) {
+		store8(SP, (1u << 8) | P);
+	} else {
+		store16(SP, P);
+	}
+	SP--;
+	setFlag(b, 0);
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHX() {
-	// TODO
+	// TODO TO CHECK
+	if (indexRegistersAre8Bit()) {
+		store8(SP, static_cast<Byte>(X.load()));
+	}
+	else {
+		store16(SP, X.forceLoadFull());
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePHY() {
-	// TODO
+	// TODO TO CHECK
+	if (indexRegistersAre8Bit()) {
+		store8(SP, static_cast<Byte>(Y.load()));
+	}
+	else {
+		store16(SP, Y.forceLoadFull());
+	}
+	SP--;
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePLA() {
-	// TODO
+	// TODO TO CHECK
+	SP++;
+	if (memoryAndAccumulatorAre8Bit()) {
+		A = load8(SP);
+	}
+	else {
+		A = load16(SP);
+	}
+	setZeroNegFlags(A);
 	return 0;
 };
 
 Blaze::Cycles Blaze::CPU::executePLB() {
-	// TODO
+	// TODO TO CHECK
+	SP++;
+	if (e != 0) {
+		DBR = load8(SP);
+		setFlag(n, (1u << 7) && DBR != 0);
+	} else {
+		DBR = static_cast<Byte>(load16(SP));
+		setFlag(n, (1u << 15) && DBR != 0);
+	}
```

Comment:
As noted in `PHB`, the data bank register is always 8 bits wide.

Also, you wrote `(1u << 7) && DBR != 0` and `(1u << 15) && DBR != 0`, but I think you meant `((1u << 7) & DBR) != 0` and `((1u << 15) & DBR) != 0`. The first versions are effectively the same as `DBR != 0` since `(1u << 7)` and `(1u << 15)` are both non-zero and the `&&` operator converts its operands to booleans.